src/components/app/MenuButtons/Download.js

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import { compress } from '../../../utils/gz';
+import ArrowPanel from '../../shared/ArrowPanel';
+import ButtonWithPanel from '../../shared/ButtonWithPanel';
+import { serializeProfile } from '../../../profile-logic/process-profile';
+import prettyBytes from '../../../utils/pretty-bytes';
+import { sendAnalytics } from '../../../utils/analytics';
+
+import type { StartEndRange } from '../../../types/units';
+import type { Profile } from '../../../types/profile';
+
+require('./Download.css');
+
+type Props = {
+  profile: Profile,
+  rootRange: StartEndRange,
+};
```

Comment:
nit: can you make these props exact and read only ? I've seen the same issue in other components below too. I know it wasn't necessarily the case in the existing code, but this is a good moment to do it, especially that it's not painful.